crates/wash-cli/src/app/mod.rs

```diff
@@ -216,6 +220,18 @@ async fn deploy_model(cmd: DeployCommand) -> Result<DeployModelResponse> {
         None => load_app_manifest("-".parse()?).await?,
     };
 
+    // If --replace was specified, we should attempt to replace the resources by deleting them beforehand
+    if cmd.replace {
+        if let (Ok(name), version) = (
+            get_model_name(&app_manifest),
+            get_model_version(&app_manifest).map(Into::into),
+        ) {
+            let _ =
```

Comment:
Nit: Might be worth logging a warning if this fails

Reply:
Ah this is a great suggestion, added!